src/java/org/apache/cassandra/index/sai/StorageAttachedIndexBuilder.java

```diff
@@ -153,12 +153,15 @@ private boolean indexSSTable(SSTableReader sstable, Set<StorageAttachedIndex> in
              LifecycleTransaction txn = LifecycleTransaction.offline(OperationType.INDEX_BUILD))
         {
             perSSTableFileLock = shouldWriteTokenOffsetFiles(sstable);
-            boolean perColumnOnly = perSSTableFileLock == null;
+            // If we were unable to get the per-SSTable file lock it means that the
+            // per-SSTable components are already being built so we only want to
+            // build the per-index components
+            boolean perIndexComponentsOnly = perSSTableFileLock == null;
```

Comment:
Nit: Maybe name it `writeIndexComponentsOnly` or `buildIndexComponentsOnly`?

Reply:
I'll go with `buildPerIndexComponentsOnly`.